diff --git a/source/i18n/dangical.cpp b/source/i18n/dangical.cpp
index 234c44b..f5343d0 100644
--- a/source/i18n/dangical.cpp
+++ b/source/i18n/dangical.cpp
@@ -136,7 +136,7 @@ static void U_CALLCONV initDangiCalZoneAstroCalc(UErrorCode &status) {
     ucln_i18n_registerCleanup(UCLN_I18N_DANGI_CALENDAR, calendar_dangi_cleanup);
 }
 
-const TimeZone* DangiCalendar::getDangiCalZoneAstroCalc(UErrorCode &status) const {
+const TimeZone* DangiCalendar::getDangiCalZoneAstroCalc(UErrorCode &status) {
     umtx_initOnce(gDangiCalendarInitOnce, &initDangiCalZoneAstroCalc, status);
     return gDangiCalendarZoneAstroCalc;
 }
diff --git a/source/i18n/dangical.h b/source/i18n/dangical.h
index 128f3af..e240272 100644
--- a/source/i18n/dangical.h
+++ b/source/i18n/dangical.h
@@ -74,7 +74,7 @@ class DangiCalendar : public ChineseCalendar {
 
  private:
 
-  const TimeZone* getDangiCalZoneAstroCalc(UErrorCode &status) const;
+  static const TimeZone* getDangiCalZoneAstroCalc(UErrorCode &status);
 
   // UObject stuff
  public: 
