--- readline.old/histfile.c	2023-02-17 13:01:18.515248292 +0300
+++ readline/histfile.c	2023-02-17 13:02:42.034252562 +0300
@@ -492,6 +492,7 @@
   tempname = 0;
   file = filename ? open (filename, O_RDONLY|O_BINARY, 0666) : -1;
   rv = exists = 0;
+  orig_lines = lines;
 
   /* Don't try to truncate non-regular files. */
   if (file == -1 || fstat (file, &finfo) == -1)
@@ -547,7 +548,6 @@
       goto truncate_exit;
     }
 
-  orig_lines = lines;
   /* Count backwards from the end of buffer until we have passed
      LINES lines.  bp1 is set funny initially.  But since bp[1] can't
      be a comment character (since it's off the end) and *bp can't be
